dev/io.openliberty.cdi.4.0.internal.services.fragment/src/io/openliberty/cdi40/internal/services/OSGIBuildServices.java

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.cdi40.internal.services;
+
+import java.util.Optional;
+
+import com.ibm.ws.kernel.service.util.ServiceCaller;
+
+import jakarta.enterprise.inject.build.compatible.spi.AnnotationBuilderFactory;
+import jakarta.enterprise.inject.build.compatible.spi.BuildServices;
+
+/**
+ *
+ */
+public class OSGIBuildServices implements BuildServices {
+
+    /** {@inheritDoc} */
+    @Override
+    public int getPriority() {
+        return 0; //default value, don't know what it should be
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public AnnotationBuilderFactory annotationBuilderFactory() {
+        Optional<AnnotationBuilderFactory> factoryOpt = ServiceCaller.callOnce(OSGIBuildServices.class, BuildServices.class,
```

Comment:
Can we add a comment here saying that `BuildServices.class` won't find `io.openliberty.cdi40.internal.services.OSGIBuildServices` because while this class implements the right interface, it is not an OSGi service itself.

Reply:
yeah will add that in a follow up PR